client/src/components/VerbContainer.jsx

```diff
@@ -62,89 +160,104 @@ export default class VerbContainer extends Component {
-        <h1>Loading Some Verbs</h1> :
-        <div>
-        <div style={{ display: 'flex', alignItems: 'center' }}>
-          <h1>{ verb.infinitive }</h1>
-          <p style={{ marginLeft: '20px' }}>({ verb.infinitiveEnglish })</p>
-          <p
-            style={{
-              border: '1px solid gray',
-              borderRadius: '8px',
-              marginLeft: '20px',
-              padding: '7px'
-            }}
-            onClick={this._nextVerb}
-          >next >></p>
-        </div>
-        <div style={{ display: 'flex', width: '100%', flexWrap: 'wrap' }}>
-          {
-            Object.keys(tenses).filter(t => tenses[t]).map(tense => (
-              <div
-                key={`${tense}${verbs[idx]}`}
-                style={{
-                  margin: '20px',
-                  marginTop: '0px',
-                  border: '1px solid #aaa',
-                  borderRadius: '8px',
-                  padding: '15px'
-                }}
-              >
-              <div
-                key={`${tense}tense`}
-                style={{
-                  fontSize: '18px',
-                  fontWeight: '600',
-                  marginBottom: '7px'
-                }}
-              >{tense}</div>
-              <ConjugationForm
-                conjugations={
-                  verbs[idx].conjugations.find(
-                    v => v.tense.toLowerCase() == tense.toLowerCase()) || {} 
-                }
-                key={`${tense}CF`}
-                tense={tense}
-                verb={verbs[idx]}
-              />
+      <StyledContainer>
+        <TenseHeaderContainer>
+          <MediaQuery query="(max-device-width: 900px)">
+            <TenseHeaderMobile 
+              activeTenses={ tenses }
+              title={ 'Tense/Mood'}
+              tenseOptions={ Object.keys(tenses) }
+              toggleTense={ this._toggleTense}
+            />
+          </MediaQuery>
+          <MediaQuery query="(min-device-width: 901px)">
+            <TenseHeader 
+              activeTenses={ tenses }
+              toggleTense={ this._toggleTense}
+            />
+            <TenseHeader 
+              activeTenses={ tenses }
+              moodSuffix={' - Subjuntivo'}
+              toggleTense={ this._toggleTense}
+            />
+          </MediaQuery>
+          
+        </TenseHeaderContainer>
+        { 
+          !hasLoaded ? 
+          <h1>Loading Some Verbs</h1> : 
+          <div>
+            <StyledVerbContainer>
+              <div id='infinitive-organizer'>
+                <div id='infinitive'>{ verb.infinitive }</div>
+                <div id='infinitive-english'>({ verb.infinitiveEnglish })</div>
               </div>
-            ))
-          }
+              <div id='next-button' onClick={ this._nextVerb }>next >></div>
+            </StyledVerbContainer>
+            <ConjugationFormList>
+            {
+              Object.keys(tenses).filter(t => tenses[t]).map((tense, tIdx) => (
```

Comment:
you put a lot of logic into the jsx. I'd pull this stuff out into methods on the class at some point to clean up the code.

Reply:
I've done that on previous React projects until someone with the opposite opinion of you told me not to. And I have to say I think I like it better this way.
I think it's easy to write a lot of `getSomethingJSX` helper functions that really end up cluttering what the `render` function means.
Here, even if the markup gets a little longer, all the rendering logic stays in the render function

Reply:
I say that if you end up having a bunch of getSomethingJSX helper functions, that is code smell and means you need to pull out more small component classes